### PR TITLE
Rust upgrade to rustc 1.5.0-dev (6217b002b 2015-09-21)

### PR DESCRIPTION
**Pull request by Manishearth**
Snapshot doesn't exist yet. Rust master+ should work.

Also https://github.com/serde-rs/aster/pull/33 hasn't merged yet.

r? @Ms2ger @larsbergstrom

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/7697)

### CODE REVIEW

**Comment by Manishearth** [comment]
> These commits modify unsafe code. Please review it carefully!
lies

File: components/servo/.cargo/config [hunk 1]
```diff
@@ -1,4 +1,4 @@
-paths = ["../../support/android-rs-glue"]
+paths = ["../../support/android-rs-glue", "/home/manishearth/Mozilla/Git/rust-aster"]
```
Comment (SimonSapin) [inline components/servo/.cargo/config]: Revert this?

File: ports/gonk/Cargo.lock [hunk 2]
```diff
@@ -1290,12 +1290,20 @@ dependencies = [
 ]
 
 [[package]]
+name = "serde"
+version = "0.6.0"
```
Comment (SimonSapin) [inline ports/gonk/Cargo.lock]: Looks like we’re using both 0.5.3 and 0.6.0 now. Can the former be removed?

File: rust-snapshot-hash [hunk 3]
```diff
@@ -1 +1 @@
-cb9323ec096dbc4b3766b770309102c7e2ed9875/rustc-1.4.0-dev
+rustc-stage2/rustc-1.5.0-dev
```
Comment (SimonSapin) [inline rust-snapshot-hash]: Waiting on a new snapshot?

**Comment by nox** [comment]
Review status: 0 of 37 files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

<sup>**[components/plugins/Cargo.toml, line 12 \[r1\]](https://reviewable.io:443/reviews/servo/servo/7697#-Jzo1K7ZIYdVHuPoVeuP)** ([raw file](https://github.com/servo/servo/blob/291fa888326e743a1e59aa7c06d5288a0643cbb7/components/plugins/Cargo.toml#L12)):</sup>
Why does the organisation fork it btw, don't we use explicit hashes anyway through Cargo.lock?

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/7697)

**Comment by Manishearth** [comment]
Review status: 0 of 37 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

<sup>**[components/plugins/Cargo.toml, line 12 \[r1\]](https://reviewable.io:443/reviews/servo/servo/7697#-Jzo1K7ZIYdVHuPoVeuP)** ([raw file](https://github.com/servo/servo/blob/291fa888326e743a1e59aa7c06d5288a0643cbb7/components/plugins/Cargo.toml#L12)):</sup>
It was a one-off thing, due to a rustup mixup.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/7697)

**Comment by Ms2ger** [comment]
Need to review updated dependencies:

* [ ] aster
* [ ] quasi
* [ ] quasi-codegen
* [ ] serde
* [ ] string_cache
* [ ] tenacious

-S-awaiting-review +S-blocked-on-external +S-awaiting-answer 

---

Reviewed 5 of 5 files at r1, 8 of 8 files at r2, 23 of 28 files at r4, 5 of 5 files at r5, 1 of 1 files at r6, 3 of 3 files at r7.
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

<sup>**[components/plugins/lib.rs, line 58 \[r6\]](https://reviewable.io:443/reviews/servo/servo/7697#-Jzq6URIFLD-rAl0k_3Z)** ([raw file](https://github.com/servo/servo/blob/452f3ace342d5758d13d89622c20f26718702725/components/plugins/lib.rs#L58)):</sup>
Curious why this is early.

---

<sup>**[components/servo/Cargo.lock, line 865 \[r6\]](https://reviewable.io:443/reviews/servo/servo/7697#-Jzq8EtEoMW6Ni-1irHh)** ([raw file](https://github.com/servo/servo/blob/452f3ace342d5758d13d89622c20f26718702725/components/servo/Cargo.lock#L865)):</sup>
Why? https://github.com/servo/rust-mozjs/pull/195 seems unrelated.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/7697)

**Comment by Manishearth** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

<sup>**[components/plugins/lib.rs, line 58 \[r6\]](https://reviewable.io:443/reviews/servo/servo/7697#-Jzq6URIFLD-rAl0k_3Z)** ([raw file](https://github.com/servo/servo/blob/452f3ace342d5758d13d89622c20f26718702725/components/plugins/lib.rs#L58)):</sup>
It could be late too. Early passes work on the AST and nothing else (and are run much earlier -- right after expansion).

Late passes have type info and stuff, but run after everything else (right before trans).

---

<sup>**[components/servo/Cargo.lock, line 865 \[r6\]](https://reviewable.io:443/reviews/servo/servo/7697#-Jzq8EtEoMW6Ni-1irHh)** ([raw file](https://github.com/servo/servo/blob/452f3ace342d5758d13d89622c20f26718702725/components/servo/Cargo.lock#L865)):</sup>
Oh, I might have updated it to get rid of the build error (which was caused by a regression in rustc).

Should I revert it?

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/7697)

**Comment by Manishearth** [comment]
Review status: 34 of 37 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

<sup>**[components/servo/Cargo.lock, line 865 \[r6\]](https://reviewable.io:443/reviews/servo/servo/7697#-Jzq8EtEoMW6Ni-1irHh)** ([raw file](https://github.com/servo/servo/blob/452f3ace342d5758d13d89622c20f26718702725/components/servo/Cargo.lock#L865)):</sup>
The linker changes I just rebased over updated this to latest mozjs anyway.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/7697)

**Comment by Ms2ger** [comment]
-S-blocked-on-external -S-awaiting-answer -S-awaiting-review -S-needs-rebase +S-needs-squash 

---

Reviewed 3 of 3 files at r8.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/7697)

**Pull request merged**